library/ui/src/main/java/com/google/android/exoplayer2/ui/TrackSelectionView.java

```diff
@@ -251,12 +258,13 @@ private void updateViews() {
     defaultView.setEnabled(true);
 
     trackGroups = mappedTrackInfo.getTrackGroups(rendererIndex);
+    sortedTrackGroups = sortedTrackGroups != null ? sortedTrackGroups : initSortedTrackGroups(trackGroups);
```

Comment:
I don't think this should be avoiding re-generating `sortedTrackGroups` if it's non-null. The `mappedTrackInfo` may have changed since this method was called, for example due to a call to `init`. Hence you always need to re-generate the `sortedTrackGroups` again.